pkg/odo/cli/url/create.go

```diff
@@ -121,7 +111,9 @@ func NewCmdURLCreate(name, fullName string) *cobra.Command {
 		},
 	}
 	urlCreateCmd.Flags().IntVarP(&o.urlPort, "port", "", -1, "port number for the url of the component, required in case of components which expose more than one service port")
-	urlCreateCmd.Flags().BoolVar(&o.urlOpenFlag, "open", false, "open the created link with your default browser")
+	_ = urlCreateCmd.MarkFlagRequired("port")
+	urlCreateCmd.Flags().StringVar(&o.componentContext, "context", "", "Use context to indicate the path where the component settings need to be saved and this directory should contain component source for local and binary components")
```

Comment:
please use `genericclioptions.AddContextFlag`. It was added in this PR https://github.com/openshift/odo/pull/1554/files